databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java

```diff
@@ -0,0 +1,422 @@
+  {
+    if (s instanceof NullColumn)
+    {
+      return null;
+    }
+    Or2AvroBasicConvert converter = fetchConverter(avroField);
+    return converter.convert(s, avroField);
+  }
+
+  private static Or2AvroBasicConvert fetchConverter(Field avroField) throws DatabusException
+  {
+    Or2AvroBasicConvert converter = convertCacheMap.get(avroField);
+    if (converter != null)
+    {
+      return converter;
+    }
+    synchronized (convertCacheMap)
+    {
+      converter = convertCacheMap.get(avroField);
+      if (converter != null)
+      {
+        return converter;
+      }
+      if (convertCacheMap.size() > maxCacheSize)
+      {
+        convertCacheMap.clear();
+      }
+      String schemaStr = avroField.schema().toString();
+      if (schemaStr.contains("int"))
+      {
+        converter = IntegerConverter.fetchInstance();
+      }
+      else if (schemaStr.contains("long"))
+      {
+        converter = LongConverter.fetchInstance();
+      }
+      else if (schemaStr.contains("double"))
+      {
+        converter = DoubleConverter.fetchInstance();
+      }
+      else if (schemaStr.contains("string"))
+      {
+        converter = StringConverter.fetchInstance();
+      }
+      else if (schemaStr.contains("bytes"))
+      {
+        converter = BytesConverter.fetchInstance();
+      }
+      else if (schemaStr.contains("float"))
+      {
+        converter = FloatConverter.fetchInstance();
+      }
+      if (converter == null)
+      {
+        throw new DatabusException("schema is " + schemaStr + ", converter not exist!");
+      }
+      convertCacheMap.put(avroField, converter);
+      return converter;
+    }
+  }
+
+  public static abstract class Or2AvroBasicConvert
+  {
+    public abstract Object convert(Column s, Field avroField) throws DatabusException;
+  }
+
+  public static class IntegerConverter extends Or2AvroBasicConvert
+  {
+    private static IntegerConverter converter = null;
+
+    public static IntegerConverter fetchInstance()
+    {
+      if (converter == null)
+      {
+        converter = new IntegerConverter();
+      }
+      return converter;
+    }
+
+    @Override
+    public Object convert(Column s, Field avroField) throws DatabusException
+    {
+      Object obj = s.getValue();
+      Integer res = null;
+      if (obj instanceof Integer)
+      {
+        res = (Integer) obj;
+      }
+      else if (obj instanceof Long)
+      {
+        res = Integer.valueOf(((Long) obj).intValue());
+      }
+      else if (obj instanceof BigDecimal)
+      {
+        res = Integer.valueOf(((BigDecimal) obj).intValue());
+      }
+      else if (obj instanceof Double)
+      {
+        res = Integer.valueOf(((Double) obj).intValue());
+      }
+      else if (obj instanceof Float)
+      {
+        res = Integer.valueOf(((Float) obj).intValue());
+      }
+      else if (obj instanceof byte[])
+      {
+        String resStr = new String((byte[]) obj, Charset.defaultCharset());
```

Comment:
Use the default character set is dangerous. Is the byte[] even possible in this case?